### ISSUE TITLE
Remote download minimal will download all input files for symlink

### ISSUE THREAD

**Issue by eagleonhill** [bazelbuild/bazel#17063]
### Description of the bug:

Symlink inputs(files they point to) are downloaded to host when remote_download_minimal is set.

Because C++ shared libraries are all symlinked to a solib directory, this causes all shared libraries to be downloaded.

I would expect them not to be downloaded. During execution, the actual file may be referenced by their digest.

```  at java.util.concurrent.locks.LockSupport.park(java.base@11.0.6/Unknown Source)
  at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:537)
  at com.google.devtools.build.lib.remote.util.Utils.getFromFuture(Utils.java:106)
  at com.google.devtools.build.lib.remote.util.Utils.getFromFuture(Utils.java:94)
  at com.google.devtools.build.lib.remote.RemoteActionInputFetcher.downloadFile(RemoteActionInputFetcher.java:149)
  at com.google.devtools.build.lib.remote.RemoteActionFileSystem.downloadFileIfRemote(RemoteActionFileSystem.java:340)
  at com.google.devtools.build.lib.remote.RemoteActionFileSystem.createSymbolicLink(RemoteActionFileSystem.java:194)
  at com.google.devtools.build.lib.vfs.Path.createSymbolicLink(Path.java:489)
  at com.google.devtools.build.lib.rules.cpp.SolibSymlinkAction.execute(SolibSymlinkAction.java:80)
  at com.google.devtools.build.lib.skyframe.SkyframeActionExecutor$5.execute(SkyframeActionExecutor.java:907)
  at com.google.devtools.build.lib.skyframe.SkyframeActionExecutor$ActionRunner.continueAction(SkyframeActionExecutor.java:1076)
  at com.google.devtools.build.lib.skyframe.SkyframeActionExecutor$ActionRunner.run(SkyframeActionExecutor.java:1031)
  at com.google.devtools.build.lib.skyframe.ActionExecutionState.runStateMachine(ActionExecutionState.java:152)
  at com.google.devtools.build.lib.skyframe.ActionExecutionState.getResultOrDependOnFuture(ActionExecutionState.java:91)
  at com.google.devtools.build.lib.skyframe.SkyframeActionExecutor.executeAction(SkyframeActionExecutor.java:492)
  at com.google.devtools.build.lib.skyframe.ActionExecutionFunction.checkCacheAndExecuteIfNeeded(ActionExecutionFunction.java:856)
  at com.google.devtools.build.lib.skyframe.ActionExecutionFunction.computeInternal(ActionExecutionFunction.java:349)
  at com.google.devtools.build.lib.skyframe.ActionExecutionFunction.compute(ActionExecutionFunction.java:169)
  at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:590)
  at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$WrappedRunnable.run(AbstractQueueVisitor.java:382)
  at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@11.0.6/Unknown Source)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@11.0.6/Unknown Source)
```
Is a stacktrace for the download.

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

Build C++ shared library with `--remote_download_minimal`

### Which operating system are you running Bazel on?

linux

### What is the output of `bazel info release`?

5.2.0

### Have you found anything relevant by searching the web?

https://github.com/bazelbuild/bazel/issues/11119 seems indicate the symlink could be cached.

**Comment by brentleyjones** [comment]
I believe 6.0 might have fixed this with https://github.com/bazelbuild/bazel/commit/32b0f5a258ee83b040d6975a6bd0795670f5cd47

**Comment by sgowroji** [comment]
As mentioned above could you please verify the same with the latest release and let us know if you still see this issue. Thanks!!

**Comment by sgowroji** [comment]
We are closing this issue as it is fixed in the above commit. Please feel free to reach us if you still have any further queries on the same. Thanks!

**Comment by sushain97** [comment]
In case it helps anyone, I verified that this is indeed fixed in Bazel 6.0.0.

Here's a minimal self-contained repro: https://github.com/sushain97/bazel-minimal-symlink-repro
